Esercizio6 completo

alicepicone opened the pull request:
implementata le eccezioni

alicepicone committed:
Modifica la tipologia di errori usando gli http errors

alicepicone committed:
refactoring

toglie get dai nomi dei metodi

alicepicone committed:
Toglie import non necessari

alicepicone committed:
Refactoring seguendo le best practices per api rest

alicepicone committed:
refactoring

toglie i verbi dall'endpoint

alicepicone committed:
Gestione delle eccezzioni personalizzate

gestisce le eccezzioni tramite controlleradvice per restituire al client i messaggi di errore legati al codice http di errore

ericPavone commented on src/main/java/com/example/demo/api/PersonController.java:
```diff
@@ -28,21 +31,21 @@ public void addPerson(@Valid @NonNull @RequestBody Person person)
-    @GetMapping("/professionByPerson")
-    public String getProfessionByPerson(@RequestParam("name") String name)
+    @GetMapping("/profession")
+    public String professionByPerson(@RequestParam("name") String name)
```
questa va sotto profession controller, se ci fai caso, identifichi la risorsa profession, e prendi la risorsa profession, con il filtro "name"

ericPavone commented on src/main/java/com/example/demo/api/ProfessionController.java:
```diff
@@ -29,7 +29,7 @@ public void addProfession(@Valid @NonNull @RequestBody Profession profession)
 
     @GetMapping
-    public List<Profession> getAllProfession()
+    public List<Profession> professions()
```
il metodo lo puoi chiamare come vuoi, non hai vincoli se non quelli delle best practices di Java, è l'endpoint che deve seguire gli standard REST di cui abbiamo parlato sta mattina

ericPavone commented on src/main/java/com/example/demo/api/PersonController.java:
```diff
@@ -28,21 +31,21 @@ public void addPerson(@Valid @NonNull @RequestBody Person person)
 
     @GetMapping
-    public List<Person> getAllPeople()
+    public List<Person> people()
```
il metodo lo puoi chiamare come vuoi, non hai vincoli se non quelli delle best practices di Java, è l'endpoint che deve seguire gli standard REST di cui abbiamo parlato sta mattina

ericPavone commented on src/main/java/com/example/demo/service/PersonService.java:
```diff
@@ -60,17 +61,19 @@ public String getNamesStartingWith(String character) {
         return sb.toString();
     }
 
-    public String getNamesByChar(String character) {
```
non usare  throws InvalidInputException, NoResultsFoundException nella signature del metodo, nonostante sia una "forzatura di java" in alcuni casi è una bad practice. 
 
 
 nei metodi che dovrebero sollevare eccezioni, usa @SneakyThrows, ad esempio  (ovviamente vatti a vedere cosa fa e aggiungilo nella roba dellgli argomenti studiati)

 
 @SneakyThrows
 public String namesByChar(String character) {} 
 
 
 PS: in questo metodo ci sta che si chiami "getNamesByChar"
 solo "namesByChar" non si capisce cosa faccia. Non è esplicativo

ericPavone reviewed (changes requested):
In generale ci siamo, oltre i commenti specifici vorrei consigliarti di usare i nomi delle exception un attimino più genererici. ad esempio: NotFoundException o BadRequestException, trattandosi di una restAPI puoi riutilizzzarle in più casi

alicepicone committed:
Trasferiti metodi del professionByPersonName

al ProfessionService e controller

alicepicone committed:
Refactoring dei nomi dei metodi

alicepicone committed:
rinominate le exception per renderle generiche

aggiunta annotazione neakyThrows per togliere le eccezzioni dalle firme dei metodi

ericPavone reviewed (changes requested):
Sai cosa fare

alicepicone committed:
Implementa il DTO per person e profession

alicepicone committed:
toglie gli id dai dto

alicepicone committed:
Risolve bug che restituiva loop di lista alla get di all people

alicepicone committed:
cambia l'endpoind delle get by id

ericPavone commented on src/main/java/com/example/demo/dto/ProfessionDTOMapper.java:
```diff
@@ -0,0 +1,16 @@
+
+import java.util.function.Function;
+
+@Service
```
@Component
